pyaedt/desktop.py

```diff
@@ -679,99 +917,140 @@ def __getitem__(self, project_design_name):
+    #     return self._version_keys
 
     @property
     def current_version(self):
         """Current AEDT version."""
-        try:
-            return self.version_keys[0]
-        except (NameError, IndexError):
-            return ""
+        return current_version()
 
     @property
-    def current_version_student(self):
+    def current_student_version(self):
         """Current student AEDT version."""
-        for version_key in self.version_keys:
-            if "SV" in version_key:
-                return version_key
-        return ""
+        return current_student_version()
+
+    @property
+    def installed_versions(self):
+        """Dictionary of AEDT versions installed on the system and their installation paths."""
+        return installed_versions()
 
-    def _init_desktop(self, non_graphical):
+    def _init_desktop(self):
+        # run it after the settings.non_graphical is set
+        self._main.pyaedt_version = pyaedtversion
         self._main.AEDTVersion = self._main.oDesktop.GetVersion()[0:6]
         self._main.oDesktop.RestoreWindow()
         self._main.sDesktopinstallDirectory = self._main.oDesktop.GetExeDir()
         self._main.pyaedt_initialized = True
-        if non_graphical or self._main.oDesktop.GetIsNonGraphical():
-            try:
-                settings.enable_desktop_logs = not self._main.oDesktop.GetIsNonGraphical()
-            except AttributeError:
-                settings.enable_desktop_logs = not non_graphical
-
-    def _set_version(self, specified_version, student_version):
-        student_version_flag = False
-        if specified_version:
-            if float(specified_version) < 2021:
-                if float(specified_version) < 2019:
-                    raise ValueError("PyAEDT supports AEDT version 2021 R1 and later.")
-                else:
-                    warnings.warn(
-                        """PyAEDT has limited capabilities when used with an AEDT version earlier than 2021 R1.
-                        PyAEDT officially supports AEDT version 2021 R1 and later."""
-                    )
-            if student_version:
-                specified_version += "SV"
-                student_version_flag = True
-            assert specified_version in self.version_keys, "Specified version {} is not known".format(specified_version)
-            version_key = specified_version
-        else:
-            if student_version and self.current_version_student:
-                version_key = self.current_version_student
-                student_version_flag = True
+
+    def _assert_version(self, specified_version, student_version):
+        # avoid evaluating the env variables multiple times
+        self_current_version = self.current_version
+        self_current_student_version = self.current_student_version
+
+        if current_version == "":
+            raise Exception("AEDT is not installed on your system. Install AEDT version 2022 R2 or higher.")
```

Comment:
You can omit the word "version" in the second sentence if you would like and just use "AEDT 2022 R2 or higher."